pgx/src/array.rs

```diff
@@ -232,20 +235,68 @@ impl ArrayPtr {
         // Most importantly, the caller has asserted this is in fact a valid [T],
         // by calling this function, so both this code and the caller can rely
         // on that assertion, only requiring that it is correct.
-        unsafe {
-            NonNull::new_unchecked(slice_from_raw_parts_mut(
-                pgx_ARR_DATA_PTR(self.at.as_ptr()).cast(),
-                len,
-            ))
-        }
+        unsafe { NonNull::new_unchecked(pgx_ARR_DATA_PTR(self.at.as_ptr()).cast()) }
     }
 }
 
 /**
 A pointer into a Postgres varlena.
 */
-pub struct RawArray<T> {
-    _ptr: ArrayPtr,
-    _len: usize,
-    ty: PhantomData<T>,
+pub struct RawArray {
+    ptr: ArrayPtr,
+    len: usize,
+}
```

Comment:
This was a cute idea for a few seconds but then you dropped the PhantomData type. These are definitely not different things.